### ISSUE TITLE
[BUG] GDI2 affects ConvertKeyToKeyCode

### ISSUE THREAD

**Issue by dein0s** [dein0s/POE-TradeMacro#1]
Merge of https://github.com/PoE-TradeMacro/POE-TradeMacro/pull/540 somehow affected `lib\ConvertKeyToKeyCode.ahk` and especially `CustomGetKeyCode()` which now cause error on on assigning keys from `AdditionalMacros.ahk`. Everything works fine on 96eb206985d048a4d93faec4383c3d1cc261d52c (right before merge)
![error](https://i.imgur.com/v9iWllV.png)
Steps to reproduce:
1) Checkout latest `beta` branch
2) Switch keyboard layout to any non-latin (chinese, korean, russian, ukrainian, belarusian, etc.)
2) Start `Run_TradeMacro.ahk`

@esunder I'll be very appriciated if you help me with this as I'm stuck right now.
I'd guess the reason is in formating numbers/dll calls, but I'm not sure exactly.

You can also find me on discord - dein0s#2248

**Comment by dein0s** [comment]
More details - seems that the script can't switch the language in with `SwitchLayout(ENG_US)` at line [76](https://github.com/PoE-TradeMacro/POE-TradeMacro/blob/beta/lib/ConvertKeyToKeyCode.ahk#L76).

**Comment by esunder** [comment]
Ok, I can probably look at this over the weekend. The bug happens regardless if GDI is enabled or not?

**Comment by esunder** [comment]
@dein0s I see a slightly different error message during application startup with Russian keyboard enabled. I could not reproduce it with Chinese simplified keyboard. It would switch the keyboard back to ENG US as the app started.

![image](https://user-images.githubusercontent.com/357754/31301160-95f146ae-aabd-11e7-911e-d94bed544cd2.png)

![image](https://user-images.githubusercontent.com/357754/31301167-9e5621fc-aabd-11e7-8f52-620f875d1236.png)

Happens multiple times for various hotkeys. Are you sure there is not something else with your beta? Have you tried backing up your repo and re-cloning?

**Comment by dein0s** [comment]
Well, this are almost the same errors (related to key converting).
Tried multiple times and on different commits, happens only after gdi was merged.

**Comment by esunder** [comment]
I just tried a checkout of main repo's master branch and deleted the PoE-ItemMacro and PoE-TradeMacro folders from my Documents folder to clear any personal settings. I get the same errors that I pasted above. This seems to be an issue going back further than my commit.

**Comment by dein0s** [comment]
It's not yet merged into master.

**Comment by dein0s** [comment]
Just to make everything clear: https://www.youtube.com/watch?v=EjyaBhE3Ixg

**Comment by dein0s** [comment]
Figured ouy with Eru that this was because of `SetFormat` caching. Still unclear what exactly triggered such behaviour after your pr, but that's not relevant for now.

**Comment by esunder** [comment]
Ok, good find!
